src/main/java/org/folio/services/domainevent/AbstractDomainEventPublisher.java

```diff
@@ -0,0 +1,136 @@
+package org.folio.services.domainevent;
+
+import static io.vertx.core.Future.succeededFuture;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static org.apache.logging.log4j.LogManager.getLogger;
+import static org.folio.rest.support.ResponseUtil.isCreateSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isDeleteSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isUpdateSuccessResponse;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.AbstractRepository;
+import org.folio.services.batch.BatchOperationContext;
+
+import io.vertx.core.Future;
+
+abstract class AbstractDomainEventPublisher<DomainType, EventType> {
+  private static final Logger log = getLogger(AbstractDomainEventPublisher.class);
+
+  protected final AbstractRepository<DomainType> repository;
+  protected final CommonDomainEventPublisher<EventType> domainEventService;
+
+  public AbstractDomainEventPublisher(AbstractRepository<DomainType> repository,
+    CommonDomainEventPublisher<EventType> domainEventService) {
+
+    this.repository = repository;
+    this.domainEventService = domainEventService;
+  }
+
+  public Function<Response, Future<Response>> publishUpdated(DomainType oldRecord) {
+    return response -> {
+      if (!isUpdateSuccessResponse(response)) {
+        log.warn("Record update failed, skipping event publishing");
+        return succeededFuture(response);
+      }
+
+      return publishUpdated(singletonList(oldRecord)).map(response);
+    };
+  }
+
+  public Function<Response, Future<Response>> publishCreated(DomainType record) {
+    return response -> {
+      if (!isCreateSuccessResponse(response)) {
+        log.warn("Item create failed, skipping event publishing");
```

Comment:
Use `Record` instead of `Item` to be consistent.

Reply:
Fixed, thanks.